openedx/core/djangoapps/content_libraries/views.py

```diff
@@ -689,3 +743,251 @@ def delete(self, request, usage_key_str, file_path):
+            return HttpResponseBadRequest('Invalid LTI login request.')
+
+
+@method_decorator(csrf_exempt, name='dispatch')
+@method_decorator(xframe_options_exempt, name='dispatch')
+class LtiToolLaunchView(TemplateResponseMixin, LtiToolView):
+    """
+    Platform tool launch view.
+
+    The launch view supports resource link launches and AGS, when enabled by the
+    platform.  Other features and resouces are ignored.
+    """
+
+    template_name = 'content_libraries/xblock_iframe.html'
+
+    @property
+    def launch_data(self):
+        return self.launch_message.get_launch_data()
+
+    def _authenticate_and_login(self, usage_key):
+        """
+        Authenticate and authorize the user for this LTI message launch.
+
+        We automatically create LTI profile for every valid launch, and
+        authenticate the LTI user associated with it.
+        """
+
+        # Check library authorization.
+
+        if not ContentLibrary.authorize_lti_launch(
+                usage_key.lib_key,
+                issuer=self.launch_data['iss'],
+                client_id=self.launch_data['aud']
+        ):
+            return None
+
+        # Check LTI profile.
+
+        LtiProfile.objects.get_or_create_from_claims(
+            iss=self.launch_data['iss'],
+            aud=self.launch_data['aud'],
+            sub=self.launch_data['sub'])
+        edx_user = authenticate(
+            self.request,
+            iss=self.launch_data['iss'],
+            aud=self.launch_data['aud'],
+            sub=self.launch_data['sub'])
+
+        if edx_user is not None:
+
+            login(self.request, edx_user)
+            perms = api.get_library_user_permissions(
+                usage_key.lib_key,
+                self.request.user)
+            if not perms:
+                api.set_library_user_permissions(
+                    usage_key.lib_key,
+                    self.request.user,
+                    api.AccessLevel.ADMIN_LEVEL)
+
+        return edx_user
+
+    def _bad_request_response(self):
+        """
+        A default response for bad requests.
+        """
+        return HttpResponseBadRequest('Invalid LTI tool launch.')
+
+    def get_context_data(self):
+        """
+        Setup the template context data.
+        """
+
+        handler_urls = {
+            str(key): xblock_api.get_handler_url(key, 'handler_name', self.request.user)
+            for key
+            in itertools.chain([self.block.scope_ids.usage_id],
+                               getattr(self.block, 'children', []))
+        }
+
+        # We are defaulting to student view due to current use case (resource
+        # link launches).  Launches within other views are not currently
+        # supported.
+        fragment = self.block.render('student_view')
+
+        return {
+            'fragment': fragment,
+            'handler_urls_json': json.dumps(handler_urls),
+            'lms_root_url': settings.LMS_ROOT_URL,
```

Comment:
Usually this should be `lms_root_url = configuration_helpers.get_value('LMS_ROOT_URL', settings.LMS_ROOT_URL)` because Open edX supports multiple "sites" where the LMS URL can be different on each site, but they share a common database.

```python
from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
```
is the import needed.